examples/tools/generate.ipynb

```diff
@@ -0,0 +1,514 @@
+    "intensity. This can be done with the\n",
+    ":func:`.generate_importance_sampled_phsp` function."
+   ]
+  },
+  {
+   "cell_type": "raw",
+   "metadata": {
+    "raw_mimetype": "text/restructuredtext"
+   },
+   "source": [
+    ".. note::\n",
+    "    Note that you still need a 'normal' phase-space sample for the\n",
+    "    :class:`.IntensityBuilderXML`. In theory, however, this shouldn't be\n",
+    "    necessary at this stage, but in the current implementation, this sample\n",
+    "    is required for computing the outermost intensity. Luckily, during data\n",
+    "    generation the parameters of the model stay fixed, so this normalization\n",
+    "    is irrelevant and just causes a constant scaling factor."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pycompwa.ui as pwa"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# specify kinematics\n",
+    "particle_list = pwa.read_particles('model.xml')\n",
+    "kinematics = pwa.create_helicity_kinematics('model.xml', particle_list)\n",
+    "\n",
+    "# speficy generators\n",
+    "generator = pwa.EvtGenGenerator(\n",
+    "    kinematics.get_particle_state_transition_kinematics_info())\n",
+    "random_generator = pwa.StdUniformRealGenerator(12345)\n",
+    "\n",
+    "# generate evenly distributed phase space sample\n",
+    "phsp_sample = pwa.generate_phsp(100000, generator, random_generator)\n",
+    "\n",
+    "# create intensity\n",
+    "intensity_builder = pwa.IntensityBuilderXML(\n",
+    "    'model.xml', particle_list, kinematics, phsp_sample)\n",
+    "intensity = intensity_builder.create_intensity()\n",
+    "\n",
+    "# generate importance-sampled phase space sample\n",
+    "phsp_sample_importance = pwa.generate_importance_sampled_phsp(\n",
+    "    2000, kinematics, generator, intensity, random_generator)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "data_sample = pwa.generate(\n",
+    "    2000, kinematics, generator, intensity, random_generator)"
+   ]
+  },
+  {
+   "cell_type": "raw",
+   "metadata": {
+    "raw_mimetype": "text/restructuredtext"
+   },
+   "source": [
+    ".. warning::\n",
+    "   The current implementation is simple and uniformly generates events in\n",
+    "   the phase space domain. Depending on the shape of your Intensity, this\n",
+    "   can be quite computation intensive. However, ideally you would only run \n",
+    "   this once for a specific reaction."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# generate data\n",
+    "data_sample = pwa.generate(\n",
+    "    100, kinematics, generator, intensity, random_generator)"
+   ]
+  },
```

Comment:
another data sample?

Reply:
Em yeah this is deff a duplicate cell :grimacing:

Reply:
https://github.com/ComPWA/pycompwa/pull/79/commits/4c318d52a08ba92ffdccf9915afabe87d5328080